src/PostgREST/Request/ApiRequest.hs

```diff
@@ -494,7 +494,21 @@ findProc qi argumentsKeys paramsAsSingleObject allProcs contentType isInvPost =
     [proc] -> Right proc
     procs  -> Left $ AmbiguousRpc (toList procs)
   where
-    matchProc = filter matchesParams $ M.lookupDefault mempty qi allProcs -- first find the proc by name
+    matchProc =
+      if null (fst findProcPartition) && isInvPost && contentType == CTApplicationJSON
+        then snd findProcPartition -- use the fallback function if none is found with the provided parameters
+        else fst findProcPartition
+    findProcPartition = overloadedProcPartition $ M.lookupDefault mempty qi allProcs -- first find the proc by name
+    -- The partition obtained has the form (overloadedProcs,fallbackProcs)
+    -- where fallbackProcs are functions with a single unnamed parameter of type json or jsonb
+    overloadedProcPartition procs = foldr select ([],[]) procs
+    select proc ~(ts,fs)
+      | matchesParams proc             = (proc:ts,fs)
+      | hasSingleUnnamedJsonParam proc = (ts,proc:fs)
```

Comment:
I like this!

Should we maybe treat the other single unnamed cases as "fallback", too?

I think that would make sense. Since in those cases, we don't have any keys from the payload to parse, the `overloadedProcs` part of the partition would be empty - and we would always fallback. So maybe this will then be `(namedProcs, singleUnnamedProcs)` instead.